ai_interface/openai_client.py

```diff
@@ -0,0 +1,33 @@
+import openai
+
+
+class OpenAiClient:
+    MODEL = "gpt-3.5-turbo"
+
+    def __init__(self, api_key, max_tokens):
+        """
+        Initializes an instance of the OpenAiClient class with the provided API key.
+        :param api_key: The API key for accessing the OpenAI service.
+        """
+        openai.api_key = api_key
+        self.max_tokens = max_tokens
+
+    def _set_prompt(self, input_data: tuple) -> str:
+        """
+        returns a prompt to send to openai based on a given input
+        :param input_data: the input that the prompt based on
+        :return: the prompt
+        """
+        raise NotImplementedError
+
+    def get_answer(self, *input_data) -> str:
```

Comment:
need to handle exceptions

Reply:
the exeptions handling will occured in the calling func